locator/token_metadata.cc

```diff
@@ -903,6 +921,14 @@ inet_address_vector_topology_change token_metadata_impl::pending_endpoints_for(c
     return endpoints;
 }
 
+std::optional<inet_address_vector_replica_set> token_metadata_impl::endpoints_for_reading(const token& token, const sstring& keyspace_name) const {
+    const auto* endpoints = maybe_migration_endpoints(endpoints_field::read_endpoints, token, keyspace_name);
+    if (endpoints == nullptr) {
+        return {};
```

Comment:
better return std::nullopt;

Reply:
why is it better?

Reply:
When returning `nullopt` it's clear that the function is returning a disengaged optional.
Otherwise one needs to look at the return type to figure what is returned.  The other return path returns a vector so this one could be mistaken to return an empty vector.

Reply:
ok